hawkbit-repository/hawkbit-repository-test/src/main/java/org/eclipse/hawkbit/repository/test/util/PostgreSqlTestDatabase.java

```diff
@@ -8,61 +8,55 @@
  */
 package org.eclipse.hawkbit.repository.test.util;
 
-import java.sql.Connection;
-import java.sql.DriverManager;
-import java.sql.PreparedStatement;
-import java.sql.SQLException;
-
 import org.apache.commons.lang3.RandomStringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.test.context.TestExecutionListener;
+import org.springframework.util.StringUtils;
 
 /**
  * A {@link TestExecutionListener} for creating and dropping MySql schemas if
  * tests are setup with MySql.
  */
 public class PostgreSqlTestDatabase extends AbstractSqlTestDatabase {
 
-    private static final Logger LOG = LoggerFactory.getLogger(PostgreSqlTestDatabase.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(PostgreSqlTestDatabase.class);
+    protected static final String POSTGRESQL_URI_PATTERN = "jdbc:postgresql://{host}:{port}/{path}?currentSchema={db}*";
 
     @Override
-    protected void createSchemaUri() {
-        schemaName = "sp" + RandomStringUtils.randomAlphanumeric(10).toLowerCase();
-        this.uri = this.uri.substring(0, uri.indexOf('?'));
-
-        System.setProperty("spring.datasource.url", uri + "?currentSchema=" + schemaName);
+    protected boolean isApplicable() {
+        return "POSTGRESQL".equals(System.getProperty("spring.jpa.database")) //
+                && MATCHER.match(POSTGRESQL_URI_PATTERN, System.getProperty(SPRING_DATASOURCE_URL_KEY))  //
+                && !StringUtils.isEmpty(getSchemaName());
     }
 
     @Override
-    protected boolean isRunningWithSql() {
-        return "POSTGRESQL".equals(System.getProperty("spring.jpa.database"));
+    protected void createSchema() {
+        final String schemaName = getSchemaName();
+        final String uri = System.getProperty(SPRING_DATASOURCE_URL_KEY);
+        LOGGER.info("\033[0;33m Creating postgreSql schema {} if not existing \033[0m", schemaName);
+
+        executeStatement(uri.split("\\?currentSchema=")[0], "CREATE SCHEMA IF NOT EXISTS " + schemaName + ";");
```

Comment:
Make use of the random generated schema name:
```
final String[] postgreUri = uri.split("\\?currentSchema=");

executeStatement(postgreUri[0], "CREATE SCHEMA IF NOT EXISTS " + postgreUri[1] + ";");
```

Reply:
Since we want to make use of `?currentSchema`, we need to get it from SPRING_DATASOURCE_URL_KEY